### PR TITLE
AG-6761 - Add label auto-rotate & skip docs

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-6761

Adds docs and an example that demonstrate how `rotation`, `autoRotate` and label skipping interact together.

### CODE REVIEW

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md [hunk 1]
```diff
@@ -313,6 +309,33 @@ will make the time axis place ticks every other month.
...
+labels.
+
+Three rotation approaches are available:
+- No rotation. X-axis labels are parallel to the axis, Y-axis labels are perpendicular.
+- Setting a fixed rotation from the axis via the `rotation` property.
+- Setting an automatically applied rotation from the axis via the `autoRotate` property. Rotation is applied if any
+  label will be wider than the gap between ticks.
+
+Label skipping is performed automatically if there is not enough space for a label to be displayed within the space
+allocated to an axis tick interval.
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md:324]: We're using the total width of all labels, shall we change the implementation to be consistent with this description?

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md [hunk 2]
```diff
@@ -313,6 +309,33 @@ will make the time axis place ticks every other month.
...
+
+If `autoRotate` is enabled, rotation will be attempted first to find a label fit, before label skipping applies.
```
[the remark this answers is not in the document]
Comment (manapeirov) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md:326]: I think `to find a label fit` is a bit misleading as the rotation angle is fixed, it implies we rotate until we find an angle that which has no overlap

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md [hunk 3]
```diff
@@ -313,6 +309,33 @@ will make the time axis place ticks every other month.
...
+Category axes have `autoRotate` enabled by default with a setting of `335`.
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md:327]: `autoRotation`

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md [hunk 4]
```diff
@@ -313,6 +309,33 @@ will make the time axis place ticks every other month.
...
+
+The following example demonstrates label rotation and skipping:
+- There is a grab handle in the bottom right to allow resizing of the chart to see how labels change with available
+  space.
+- Initially both axes have defaults applied. The X-axis is a category axis so `autoRotate` is enabled by default.
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md:332]: `autoRotation`

**Review by manapeirov — COMMENTED** [review]
Looks good just left some comments to change `autoRotate` to `autoRotation`

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md [hunk 5]
```diff
@@ -313,6 +309,33 @@ will make the time axis place ticks every other month.
...
 
+### Label Rotation & Skipping
+
+Label rotation allows a trade-off to be made between space occupied by the axis, series area, and readability of the axis
+labels.
+
+Three rotation approaches are available:
+- No rotation. X-axis labels are parallel to the axis, Y-axis labels are perpendicular.
+- Setting a fixed rotation from the axis via the `rotation` property.
+- Setting an automatically applied rotation from the axis via the `autoRotate` property. Rotation is applied if any
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md:320]: I will follow-up with a distinct PR for this.

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md (hunk 1 above)
Comment (alantreadway) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-axes/index.md]: Update the documentation to be less specific about the actual algorithm.

**Comment by alantreadway** [comment]
@manapeirov PTAL

**Review by manapeirov — APPROVED** [review]

**Pull request merged**